lib/authorize.js

```diff
@@ -0,0 +1,56 @@
+import { getLogger } from 'appium-logger';
+import { exec } from 'teen_process';
+import { getPath } from 'appium-xcode';
+import path from 'path';
+
+const logger = getLogger('AuthorizeIOS');
+
+export async function authorize (insecure) {
+
+  let xcodeDir;
+  let user;
+
+  try { 
+    // enable developer tools
+    logger.info('Enabling DevToolsSecurity');
+    await exec('DevToolsSecurity', ['--enable']);
+    // update security db -- removes authorization prompt
+    logger.info(`Updating security db for ` + 
+                `${insecure ? 'insecure' : 'developer'} access`);
+    let cmd = (`security authorizationdb write system.privilege.taskport` +
+              ` ${insecure ? 'allow' : 'is-developer'}`);
+    await exec(cmd);
+    logger.info('Granting access to built-in simulator apps');
+
+    if (!process.env.HOME) {
+      throw new Error('Could not determine your $HOME');
+    } else { 
```

Comment:
since we're `throw`ing in the `if` block, we don't need an `else` block (since throwing will short-circuit this). this is a pattern we follow. so it'd look like:

```js
if (!process.env.HOME) {
  throw new Error(...);
}
user = ...;
```